consensus/istanbul/backend/engine.go

```diff
@@ -317,29 +315,66 @@ func (sb *Backend) verifyCommittedSeals(chain consensus.ChainReader, header *typ
 		return errEmptyCommittedSeals
 	}
 
+	// Check the signatures on the current header
+	snap, err := sb.snapshot(chain, header.Number.Uint64()-1, header.ParentHash, parents)
+	if err != nil {
+		return err
+	}
 	validators := snap.ValSet.Copy()
-	// Check whether the committed seals are generated by parent's validators
+	err = sb.checkValidatorSignatures(header.Hash(), validators, extra.Bitmap, extra.CommittedSeal)
+	if err != nil {
+		return err
+	}
 
-	proposalSeal := istanbulCore.PrepareCommittedSeal(header.Hash())
-	// 1. Get committed seals from current header
-	myValidatorIndex, myValidator := validators.GetByAddress(sb.Address())
+	if len(parents) > 1 {
```

Comment:
Why do we only check the parent signatures if `len(parents) > 1`? Will there be cases where `parents` is empty? IIRC, there's no guarantee that parents will actually be populated

Reply:
if the length of parents == 0, then we're at the genesis block, and there's no parent
if the length is 1, then the parent is going to be the genesis block which validators never signed on
if the length is 2, then the parent is going to be the block #1, which was a block generated normally via the consensus process, which will have validator sigs, so only then does it start to make sense to check the signatures.

I could make this check `if block.Number.Uint64() > 1` if you'd like

Reply:
I strongly agree with @asaj 's concern as well.  The `VerifiyHeader` function in `engine.go` will ultimately pass a `nil` value for parents.